perceval/simulators/simulator_factory.py

```diff
@@ -137,5 +137,6 @@ def build(circuit: ACircuit | Processor | list,
 
         if convert_to_circuit:
             circuit = _unitary_components_to_circuit(circuit, m)
-        simulator.set_circuit(circuit)
+        # m can only be retrieved from a Processor (and useful only in that case)
+        simulator.set_circuit(circuit, m if m else None)
```

Comment:
I think `m` will never be `None` here, it is set to 0 at the beginning of this method (l73 has `m=0`)

Reply:
Yes, but the methods in the set_circuit need `m` to be `None` if it is not defined, whereas the `_unitary_components_to_circuit` method needs m to be 0